File: Core/Entities/Product.cs
namespace Core.Entities
{
    public class Product : BaseEntity
    {
        public string? ItemCode { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public int Price { get; set; }
        public string? ImageName { get; set; } 
        public string? ShortDescription { get; set; }
        public string? Tags { get; set; }
        public bool Active { get; set; }

        // Include Properties 
        public Category? Category { get; set; }
        public int CategoryID { get; set; }
        public Brand? Brand { get; set; }
        public int BrandID { get; set; }
        public Branch Branch { get; set; }
        public int BranchID { get; set; }

    }
}

File: Ecart/src/app/shared/models/product.ts
import {commonParams} from "./commonparams"

export interface IProduct {
    pageIndex: number;
    pageSize: number;
    count: number;
    records: IProductList[];
};

export interface IProductList{
    id: number;
    name: string;
    description: string;
    price: number;
    imageName: string;
    shortDescription: string;
    tags: string;
    active: boolean;
    itemCode: string;
    brandID: number;
    categoryID: number;
    branchID: number;
    category: string;
    brand: string;
    branch: string;
};

export class productParams extends commonParams{
    brandId? : number=null;
    categoryID? :number=null;
    branchID?:number =null;
    itemCode? :string;
}


